zeebe/gateway-grpc/src/main/java/io/camunda/zeebe/gateway/interceptors/impl/IdentityTenantService.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.interceptors.impl;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import io.camunda.identity.sdk.Identity;
+import io.camunda.identity.sdk.tenants.dto.Tenant;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import org.jetbrains.annotations.NotNull;
+
+public class IdentityTenantService {
+
+  private final RejectedExecutionException ree;
+
+  private final LoadingCache<String, List<Tenant>> tenantCache;
+  private final Semaphore semaphore;
+  private final Identity identity;
+
+  private final int semaphoreCapacity = 300;
+  private final long semaphoreTimeout = 1;
+  private final long cacheSize = 1000;
```

Comment:
❓ How did you decide on the size? It feels like 1000 is a lot, but then again maybe it isn't. I've got no clue 😅

Reply:
That's a good point, I didn't have a good gauge on what would be a good value here. The cache size is directly correlated to the number of users, i.e. Identity JWT tokens, and that number may vary significantly per Zeebe user.

I agree, 1000 seems like a very large number for JWT tokens. I'll try to poll the Identity team for some metrics so I can derive some sensible number.

Reply:
Well, we can also see it from a memory point of view. Can we estimate how much 1000 entries would take? If it's still low, than that's fine I think.

Reply:
I'm not sure it is. We're storing a list of tenants, so if there is a large number of tenants (let's say 200-300?) mapped to 1000 JWT tokens, is this still a low memory footprint?

Reply:
Considering the average size of a JWT string (cache key) is ~1KB, and a single `tenantId` with max size (32 UTF-8 chars) is ~56bytes, plus the overhead of the Cache objects (let's say ~32 bytes if it's similar to a `Map.Entry`), a single entry in the cache would be ~1,1KB.

- Let's make an assumption that an installation has ~50 tenants (that's still quite a large number). If they are all assigned to a single JWT token, that's 50x56bytes=2,8KB.
- For 1000 entries where all JWT tokens are assigned to all tenants: 1000x(1,1KB+2,8KB+32bytes)=**3,9 MB**

I'll drop the default size to 200 entries. It's still a large number of JWT tokens, but the memory footprint should be ~780 KB then.